client/src/helpers/unwrapCourse.jsx

```diff
@@ -0,0 +1,46 @@
+import { Button } from "@mui/material";
+import React from "react";
+
+const COURSE_CODE_PATTERN = /([A-Z]{2,4}\s[0-9]{3})/g;
+const START_MARKER_PATTERN = /\$\[/g;
+const END_MARKER_PATTERN = /\]\$/g;
+
+/**
+ * Replaces course codes with searchable buttons, and remove the markings
+ *
+ * @param {course} marked string representing prerequisites or dependencies
+ * @param {courseHandler} courseHandler function used to search for a course when clicked on it
+ * @returns an unmarked string representing prerequisites or dependencies, where courses are replaced
+ *        with re-searchable buttons
+ */
+export function unwrapCourse(course, courseHandler) {
+  // create an array from the string, where where even-numbered elements are non-course code parts,
+  // and odd-numbered elements are course codes
+  const parts = course.split(COURSE_CODE_PATTERN);
+
+  return parts.map((part, index) => {
+    if (index % 2 === 1) {
+      // Wrap course code with searchable button
+      return (
+        <React.Fragment key={index}>
+          <Button
+            variant="contained"
+            size="small"
+            onClick={() => courseHandler(part)}
+          >
+            <h5>{part}</h5>
+          </Button>
+        </React.Fragment>
+      );
+    } else {
+      return (
+        // Remove markings
+        <React.Fragment key={index}>
```

Comment:
Why are there start and end markers in non-course code parts?

Reply:
The regex pattern I used matches only the course code and places it in every even-numbered index. Therefore, the markers are left in the odd-numbered indices/non-course-code parts.